### PR TITLE
Redesign k8

### PR DESCRIPTION
**Pull request by l0k0ms**
Preview:
https://docs-staging.datadoghq.com/gus/redesign-k8/integrations/kubernetes/

This PR merge kubernetes_state and kubernetes as one big integration.

It explains for both integration how to do it from the host or from a containers.
It explains how to set permission with RBAC enabled

It moves all not-configuration related content to FAQ section

### CODE REVIEW

File: content/integrations/faq/gathering-kubernetes-events.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+---
+title: Gathering kubernetes events
```
Comment (irabinovitch) [inline content/integrations/faq/gathering-kubernetes-events.md]: *Kubernetes

File: content/integrations/faq/gathering-kubernetes-events.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+kind: faq
+customnav: integrationsnav
+---
+
+As the 5.17.0 release, Datadog Agent now supports built in leader election option for the Kubernetes event collector.
```
Comment (irabinovitch) [inline content/integrations/faq/gathering-kubernetes-events.md]: @hkaj Can we make this the default behavior starting in 5.21+ ?

File: content/tracing/faq/i-want-to-instrument-a-language-you-don-t-yet-support-how-does-the-api-work.md [hunk 3]
```diff
@@ -5,4 +5,4 @@ autotocdepth: 3
 customnav: tracingnav
 ---
 
-The API is open source and available on GitHub here: https://github.com/DataDog/dd-trace-py/blob/master/ddtrace/api.py Documentation is coming but in the meantime please direct questions to our [help channels](/help).
\ No newline at end of file
+The API is open source and available on GitHub [here](https://github.com/DataDog/dd-trace-py/blob/master/ddtrace/api.py).  Documentation is coming but in the meantime please direct questions to our [help channels](/help).
```
Comment (irabinovitch) [inline content/tracing/faq/i-want-to-instrument-a-language-you-don-t-yet-support-how-does-the-api-work.md]: I dont think the API is open source. The agent and receiver are?

@jeremy-lq can you help with wording here?

File: content/integrations/kubernetes_state.md [hunk 4]
```diff
@@ -49,3 +31,29 @@ The Kubernetes-state check does not include any event at this time.
...
+
+You can also contact our Support Team via email at [support@datadoghq.com](mailto:support@datadoghq.com).
+
+### Over Slack
+
+Reach out to our team and other Datadog users on [Slack](http://chat.datadoghq.com/).
+
+## Further Reading
+
+* [How to get more out of your Kubernetes integration?](/agent/faq/how-to-get-more-out-of-your-kubernetes-integration)
```
Comment (irabinovitch) [inline content/integrations/kubernetes_state.md]: The title here seems wrong. Maybe Collecting additional metrics from cAdvisor

File: content/integrations/kubernetes_state.md [hunk 5]
```diff
@@ -49,3 +31,29 @@ The Kubernetes-state check does not include any event at this time.
...
+* [How to report host disk metrics when dd-agent runs in a docker container?](/agent/faq/how-to-report-host-disk-metrics-when-dd-agent-runs-in-a-docker-container)
```
Comment (irabinovitch) [inline content/integrations/kubernetes_state.md]: dd-agent -> Datadog Agent

File: content/integrations/kubernetes_state.md [hunk 6]
```diff
@@ -49,3 +31,29 @@ The Kubernetes-state check does not include any event at this time.
 
 ### Service Checks
 The Kubernetes-state check does not include any service check at this time.
+
+## Troubleshooting
+
+If you have any questions about Datadog or a use case our [Docs](https://docs.datadoghq.com/) didn’t mention, we’d love to help! Here’s how you can reach out to us:
```
Comment (irabinovitch) [inline content/integrations/kubernetes_state.md]: Aren't we dropping all the boilerplate support links?

File: content/integrations/kubernetes.md [hunk 7]
```diff
@@ -11,16 +11,276 @@ description: "{{< get-desc-from-git >}}"
 {{< img src="integrations/kubernetes/k8sdashboard.png" alt="Kubernetes Dashboard" responsive="true" >}}
```
Comment (irabinovitch) [inline content/integrations/kubernetes.md]: So we're not parsing readmes here?

File: content/integrations/faq/why-is-there-a-container-in-each-kubernetes-pod-with-0-cpu-and-minimal-disk-ram.md [hunk 8]
```diff
@@ -0,0 +1,9 @@
+---
+title: Why is there a container in each Kubernetes pod with 0% CPU and minimal disk/ram?
+kind: faq
+customnav: integrationsnav
+---
+
+These are pause containers (docker_image:gcr.io/google_containers/pause.*) that K8s injects into every pod to keep it populated even if the "real” container is restarting/stopped.
```
Comment (irabinovitch) [inline content/integrations/faq/why-is-there-a-container-in-each-kubernetes-pod-with-0-cpu-and-minimal-disk-ram.md]: This isn't relevant as we ignore pause containers by default

File: content/integrations/kubernetes_state.md (hunk 4 above)
Comment (l0k0ms) [reply content/integrations/kubernetes_state.md]: This article is merged in main k8 article, he will be removed from the repo soon

File: content/integrations/kubernetes_state.md (hunk 6 above)
Comment (l0k0ms) [reply content/integrations/kubernetes_state.md]: yup, this article is going to be removed

File: content/integrations/kubernetes.md (hunk 7 above)
Comment (l0k0ms) [reply content/integrations/kubernetes.md]: no, this article is to complex to be build on the fly from repos

File: content/integrations/faq/why-is-there-a-container-in-each-kubernetes-pod-with-0-cpu-and-minimal-disk-ram.md (hunk 8 above)
Comment (l0k0ms) [reply content/integrations/faq/why-is-there-a-container-in-each-kubernetes-pod-with-0-cpu-and-minimal-disk-ram.md]: 👍

**Pull request merged**